core/dbt/cli/flags.py

```diff
@@ -355,6 +355,8 @@ def fire_deprecations(self):
         # not get pickled when written to disk as json.
```

Comment:
I'd like to refactor the deprecated_env_var_warnings mechanism above to use the new deprecations.buffer pattern but will defer that to a follow-up PR